### ISSUE TITLE
Discussion: The AI-driven JavaScript tutorial

### ISSUE THREAD

**Issue by iliakan** [javascript-tutorial/en.javascript.info#3587]
Hello!

Let's discuss the new generation of the tutorial, that is yet to come.
I've been actively experimenting with generative AI lately, and it's definitely helpful in explanations. 

To put it simple, generative AI is a mathematical algorithm, that consumed a LOT of information from the internet and, based on it, can respond to questions, solve tasks (including those on this website), generate code, etc. 

It doesn't replace humans totally, and also doesn't make learning obsolete, as humans still need to guide it, understand the generated code to fix it where it's wrong.

Generative AI can be a great helper, especially for the basic topics, which are very well covered on the internet (and so AI has digested a lot of information and became competent).

How do you imagine the AI-driven Javascript Tutorial?

**Comment by joaquinelio** [comment]
Sorry the length! 
I got too excited.
Should I rewrite it?
May the ai can get a summary

**Comment by iliakan** [comment]
Last year was tough for me, but the situation gradually changes to better.
I'm going to devote much more time to this project, so let's imagine what it can be! ;)

**Comment by joaquinelio** [comment]
Glad to hear that both, human and project, are alive 🚀 
I don't see many options in the interface
A noticeable icon in the home page working mostly as a smart site search,
A discrete and always visible icon in the rest.

How I imagine it
A chat panel showing from the bottom in my phone,,
or from the right in my notebook,
a istinctive AI font style :)

Some ideas
You can instruct AI to consider the current user's view as a parameter, that gives it a context if the question is too ambiguous. With time a user learn to refine the questions but it would help beginners who lack the jargon.

AI is freaking smart, unlike humans, it seems it considers all the clues you give it.
So you could also tune the response if it knows something about the user. You may put some config, asking if it's a begginer, a veteran or the background knoledge ( I found useful answers when I asked for comparisons, "how is different from x lang")

AI excels in grammar, it helps a lot to the understanding. but the problem I often find, mostly outside programing world, it can be very wrong and very, VERY convincing. So can humans. nothing new here, we know that. Lets play then.

**Comment by joaquinelio** [comment]
woa!
Just ask AI
It just gave better answer than mine

**Comment by omentic** [comment]
> How we can use AI to improve the tutorial? Share your ideas! ;)

Please do not. :(

I use javascript.info because it is a *reference*. It is right. Introducing AIs into the mix means it can be *wrong*.

That would make it useless as a reference. I'm really disappointed MDN has been pushing their "AI Help", despite the overwhelming backlash it got on launch. I do not want to see more excellent references go the same way.

If people desire different, wordier, *incorrect* explanations, then that is their prerogative, but please keep it out of this.

**Comment by smith558** [comment]
I noticed there hadn't been much activity within the repo recently (in terms of merged PRs, resolved issues, etc.) and it has decreased noticeably. Is the project still ongoing? Are you perhaps looking for additional maintainers?

Regarding an AI use case in tutorials/reference style projects, sentry.io have an AI chatbot integrated into their documentation https://docs.sentry.io/platforms/javascript/.

**Comment by smith558** [comment]
> > How we can use AI to improve the tutorial? Share your ideas! ;)
> 
> Please do not. :(
> 
> I use javascript.info because it is a _reference_. It is right. Introducing AIs into the mix means it can be _wrong_.
> 
> That would make it useless as a reference. I'm really disappointed MDN has been pushing their "AI Help", despite the overwhelming backlash it got on launch. I do not want to see more excellent references go the same way.
> 
> If people desire different, wordier, _incorrect_ explanations, then that is their prerogative, but please keep it out of this.

I _mostly_ share this sentiment too.

**Comment by joaquinelio** [comment]
Dont underestimate AI nor students!
Everyone needs to jump on the revolution. Not just because.

**Wrong answers**
The problem is not only wrong data, 
the problem it is VERY CONVINCING

But you need to learn that, not avoid it. Be always dubious and ask again, giving context and the socratic method works very well. Fortunately it doesn't pass turing test yet, eventually you can figure its fail or yours.

I would put, in addition to the positronic brain icon, a warning and some recommendations like before. That's all.

**Wordiness**
Not just wordy, excellent grammar and construction.
That helps a lot. It eliminates ambiguity, it put concepts the weight they need.

**Planner**
you can instruct ai to help in your path. I thought this site wasn't suitable due the lack of tests (compared to camps) but I realized you can instruct ai to figure it out. It can ask you some questions (like 20 questions game) then adapt to your (honest) answers and make your profile from that. Yes, it would be better if it  suggests chapters to read rather than trying to explain all by itself.

**Tutorial improvement**
I relied in disqus comments to understand what's wrong., grammar, concepts, translation.
For example, Mocha resulted very frustrating for begginers. It shouldn't be! So I put a PR 
I think will address that.
With Ai, the main maintainer can just ask ai what are students struggling with.

God save our new ruler. 🚀 😉

**Comment by stevenseb** [comment]
Perhaps better than adding an AI chatbot to javascript.io it would be useful to create a new section of the tutorial that provides guidance and examples of the best practices for using certain already developed AI tools in order to increase the efficiency of coding. Perhaps this could relate specifically to ways that AI tools can be used to speed the process of coding in JS, debugging in JS or maybe even performing code testing with AI. There are already plenty of newly developed AI tools and people should learn how to make the most from those while coding JS. True, it might be tempting to create an AI JS tutor that learners can use in the learning process.

**Comment by joshua-goldstein** [comment]
First of all, thank you for the high quality tutorial.  The qualities I appreciate most in your writing are conciseness and correctness.  In my (admittedly limited) experience with ChatGPT, all too often it provides wordy explanations that are only half true.  If it helps you develop a preliminary draft of an article, that of course is not my business, but I cannot image its quality of writing will match yours without significant revision.

**Comment by Wariz007** [comment]
I don't think we should disregard AI. Wether we like it or not AI is here to stay. 

What we should do is find a way to optimise it and mitigate its errors like when it provide wrong answers. 

Avoiding AI because it is sometimes wrong will just set you back while people that use it will continue to move forward.
